msg_test.go

```diff
@@ -213,6 +213,31 @@ func TestUnpackDomainName(t *testing.T) {
 	}
 }
 
+func TestPackDomainNameCompressionMap(t *testing.T) {
```

Comment:
We need another compression test?

Reply:
I think so considering this was subtly broken before. See 5547fd63a0ca32a7aa0db0b0cff51fb96fa4df6c & 6aa05940d56cc6879e519112fcf68b9b486b1d6f. I'd definitely like to keep this here because it's an easy bug to reintroduce.